orca-clouddriver/src/main/java/com/netflix/spinnaker/orca/clouddriver/tasks/monitoreddeploy/MonitoredDeployBaseTask.java

```diff
@@ -267,22 +263,21 @@ private MonitoredDeployStageData getStageContext(StageExecution stage) {
     return stage.mapTo(MonitoredDeployStageData.class);
   }
 
-  private String getRetrofitLogMessage(Response response) {
-    if (response == null) {
+  private String getRetrofitLogMessage(SpinnakerServerException spinnakerException) {
+    if (!(spinnakerException instanceof SpinnakerHttpException)) {
       return "<NO RESPONSE>";
     }
 
+    SpinnakerHttpException httpException = (SpinnakerHttpException) spinnakerException;
     String body = "";
     String status = "";
     String headers = "";
+    ObjectMapper objectMapper = new ObjectMapper();
 
     try {
-      status = String.format("%d (%s)", response.getStatus(), response.getReason());
-      body =
-          CharStreams.toString(
-              new InputStreamReader(response.getBody().in(), StandardCharsets.UTF_8));
-      headers =
-          response.getHeaders().stream().map(Header::toString).collect(Collectors.joining("\n"));
+      status = String.format("%d (%s)", httpException.getResponseCode(), httpException.getReason());
+      body = objectMapper.writeValueAsString(httpException.getResponseBody());
```

Comment:
Need a null check for httpException.getResponseBody.

Reply:
Added null check with the help of Optional<> class in java